packages/x-charts/src/BarChart/useBarChartProps.ts

```diff
@@ -95,6 +95,13 @@ export const useBarChartProps = (props: BarChartProps) => {
       })),
     [hasHorizontalSeries, series],
   );
+
+  const defaultXAxis = hasHorizontalSeries ? undefined : defaultBandXAxis;
+  const processedXAxis = React.useMemo(
+    () => (xAxis ? xAxis.map((axis) => ({ scaleType: 'band' as const, ...axis })) : defaultXAxis),
+    [defaultXAxis, xAxis],
+  );
```

Comment:
If I get it correctly, this causes issues with the following config. Even if the layout is horizontal, the xAxis will get a band scale, which is not the expected behavior

```jsx
<BarChart
  xAxis={[{ min: 0, max: 100 }]}
  layout="horizontal"
  {...}
/>
```

Reply:
Good point. Didn't think of the horizontal case. Will fix